### PR TITLE
refactor: use jsdom instead of xmldom

### PR DESCRIPTION
**Pull request by tolujimoh**
To select elements in the DOM document by query (e.g. `img[image_type=graphic]`) we need to use [querySelectorAll](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelectorAll) method.

[xmldom](https://github.com/jindw/xmldom) which is used by syndication api to [parse the HTML string to a DOM Document](https://github.com/Financial-Times/next-syndication-api/blob/88e39c42aeffc8f3a95e76c9b8c58793140dd609/server/lib/format-article-xml.js#L12) uses [W3C DOM Level 2](https://www.w3.org/TR/2000/REC-DOM-Level-2-Core-20001113/core.html) which doesn't have querySelectorAll method in the Document Object.

[jsdom](https://github.com/jsdom/jsdom) uses [WHATWG DOM and HTML Standards](https://dom.spec.whatwg.org/) which has querySelectorAll method in the Document Object.

As such, this PR replaces xmldom with jsdom.

### CODE REVIEW

File: server/lib/decorate-article.js [hunk 1]
```diff
@@ -5,15 +5,15 @@ const path = require('path');
...
 const moment = require('moment');
-const { DOMParser } = require('xmldom');
+const { JSDOM } = require('jsdom');
 
 const Handlebars = handlebars();
 
 const BASE_PATH = path.dirname(path.relative(process.cwd(), __dirname));
-const HD = Handlebars.compile(fs.readFileSync(path.resolve(BASE_PATH, './views/partial/article_metadata_hd.html.hbs'), 'utf8'), { noEscape: true });
-const FT = Handlebars.compile(fs.readFileSync(path.resolve(BASE_PATH, './views/partial/article_metadata_ft.html.hbs'), 'utf8'), { noEscape: true });
+const headerTemplate = Handlebars.compile(fs.readFileSync(path.resolve(BASE_PATH, './views/partial/article_metadata_hd.html.hbs'), 'utf8'), { noEscape: true });
```
Comment (thurstontye) [inline server/lib/decorate-article.js:13]: much better, easier to read

File: server/lib/enrich/article.js [hunk 2]
```diff
@@ -54,12 +54,12 @@ module.exports = exports = function article(content, format) {
 
 		content.document = decorateArticle(content.document, content);
 
-		content.bodyHTML__CLEAN = content.document.toString();
+		content.bodyHTML__CLEAN = content.document.body.outerHTML;
```
Comment (thurstontye) [inline server/lib/enrich/article.js:57]: Is this definately equivalent ot document.toString()?

File: server/lib/decorate-article.js [hunk 3]
```diff
@@ -5,15 +5,15 @@ const path = require('path');
...
+const footerTemplate = Handlebars.compile(fs.readFileSync(path.resolve(BASE_PATH, './views/partial/article_metadata_ft.html.hbs'), 'utf8'), { noEscape: true });
```
Comment (AniaMakes) [inline server/lib/decorate-article.js:14]: Really good idea to update the names, makes it much clearer! 🍪

File: server/lib/enrich/article.js [hunk 4]
```diff
@@ -54,12 +54,12 @@ module.exports = exports = function article(content, format) {
...
 
 		// we need to strip all formatting — leaving only paragraphs — and pass this to pandoc for plain text
 		// otherwise it will uppercase the whole article title and anything bold, as well as leave other weird
 		// formatting in the text file
-		content.bodyHTML__PLAIN = toPlainText(content.document.toString());
+		content.bodyHTML__PLAIN = toPlainText(content.document.body.outerHTML);
```
Comment (thurstontye) [inline server/lib/enrich/article.js:62]: Why not do `content.bodyHTML__PLAIN = toPlainText(content.bodyHTML__CLEAN);`?

**Review by thurstontye — APPROVED** [review]
LGTM 👍

File: server/lib/get-word-count.js [hunk 5]
```diff
@@ -2,8 +2,8 @@
 
 const RE_VALID_ONE_CHARS = /^[IAa0-9]$/;
```
Comment (AniaMakes) [inline server/lib/get-word-count.js:4]: Given that in that in another PR your replaces the home brew function, is this file actually doing anything?

**Review by AniaMakes — COMMENTED** [review]
This looks OK to me, but this is on "reading this, it makes sense". I guess as long as tests pass, then it works? 😅

File: server/lib/enrich/article.js (hunk 4 above)
Comment (tolujimoh) [inline server/lib/enrich/article.js:62]:
```suggestion
		content.bodyHTML__PLAIN = toPlainText(content.bodyHTML__CLEAN);
```

**Pull request merged**